ldms/src/sampler/llnl/edac.c

```diff
@@ -390,15 +390,20 @@ static int sample(struct ldmsd_sampler *self)
 		myFile = fopen(command[i], "r");
 		if (myFile == NULL)
 		{
-			msglog(LDMSD_LERROR, SAMP ": failed to open file\n");
+			msglog(LDMSD_LERROR, SAMP ": failed to open file %s\n",
+					command[i]);
 			rc = EINVAL;
 			edac_valid=0;
 			goto out;
 		}
+		errno = 0;
 		s = fgets(lineBuffer, sizeof(lineBuffer), myFile);
+		int en = errno;
```

Comment:
I'm a bit stumped by this paradigm. Why not just use errno instead of assigning it to a local?

Reply:
Can one reasonably assume that msglog arg setup is not going to mess with errno before the argument at the position of en is used? If so, then you are correct that en is unnecessary and is in this case just seat-belt and suspenders.